File: api/src/main/java/org/delivery/api/domain/token/business/TokenBusiness.java
package org.delivery.api.domain.token.business;

import lombok.RequiredArgsConstructor;
import org.delivery.common.annotation.Business;
import org.delivery.common.error.ErrorCode;
import org.delivery.common.exception.ApiException;
import org.delivery.api.domain.token.controller.model.TokenResponse;
import org.delivery.api.domain.token.convertor.TokenConverter;
import org.delivery.api.domain.token.service.TokenService;
import org.delivery.db.user.UserEntity;

import java.util.Optional;

@RequiredArgsConstructor
@Business
public class TokenBusiness {

    private final TokenService tokenService;
    private final TokenConverter tokenConverter;

    /**
     * 1. userEntity -> userId
     * 2. access, refresh token 발급
     * 3. convertor -> response
     * @param userEntity
     * @return
     */
    public TokenResponse issueToken(UserEntity userEntity){



        return Optional.ofNullable(userEntity)
                .map(ue -> {
                    return ue.getId();
                })
                .map(userId -> {
                    var accessToken = tokenService.issueAccessToken(userId);
                    var refreshToken = tokenService.issueRefreshToken(userId);
                    return tokenConverter.toResponse(accessToken,refreshToken);
                })
                .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }

    public Long validationToken(String accessToken){
        var userId = tokenService.validationToken(accessToken);
        return userId;
    }
}

File: account/src/main/kotlin/org/delivery/account/domain/token/business/TokenBusiness.kt
package org.delivery.account.domain.token.business

import org.delivery.account.domain.token.controller.model.TokenValidationRequest
import org.delivery.account.domain.token.controller.model.TokenValidationResponse
import org.delivery.account.domain.token.service.TokenService
import org.delivery.common.annotation.Business

@Business
class TokenBusiness (
    private val tokenService : TokenService,
){

    fun tokenValidation(tokenValidationRequest: TokenValidationRequest?)
    :TokenValidationResponse?{
        tokenService.validationToken(tokenValidationRequest?.tokenDto?.token).let{
            return TokenValidationResponse(
                userId = it
            )
        }
    }
}
